src/lib.rs

```diff
@@ -59,31 +59,26 @@ include!(concat!(env!("OUT_DIR"), "/build_rev.rs"));
 
 /// A .nix file.
 #[derive(Hash, PartialEq, Eq, Clone, Debug, Serialize, Deserialize)]
-pub struct NixFile(PathBuf);
+pub enum NixFile {
+    /// A .nix file which describes a shell environment
+    Shell(PathBuf),
+    /// A .nix file which describes a list of services
+    Services(PathBuf),
+}
```

Comment:
This doesn’t look like the right enum to put this in, rather there should be a second enum with two `NixFiles` in the fields.

Reply:
The way I thought about this is simply that lorri deals with two types of nix files, namely ones that describe a shell and ones that describe services. The `NixFile` enum describes exactly that.

Reply:
I guess we can go both ways.